runtime-sdk/modules/evm/src/backend.rs

```diff
@@ -471,8 +486,13 @@ impl<'ctx, 'backend, 'config, C: TxContext, Cfg: Config> StackState<'config>
     }
 
     fn inc_nonce(&mut self, address: H160) -> Result<(), ExitError> {
-        // Do not increment the origin nonce as that has already been handled by the SDK.
-        if address == self.origin() {
+        let ctx = self.backend.ctx.borrow_mut();
+
+        // Do not increment the origin nonce as that has already been handled by the SDK. But do
+        // record that the nonce should be incremented based on EVM semantics so we can adjust any
+        // results from the `basic` method.
+        if address == self.origin() && !ctx.is_simulation() {
+            self.substate.origin_nonce_incremented = true;
```

Comment:
Should this be a counter?

Reply:
No, as the nonce is only incremented once. This is basically because the EVM assumes it is the entire runtime handling also transaction logic. 

But in our case the SDK (e.g. the `accounts` module) is handling nonce increments.